modules/core/test/test_mat.cpp

```diff
@@ -1804,4 +1804,24 @@ TEST(Mat_, from_initializer_list)
     ASSERT_DOUBLE_EQ(norm(A, B, NORM_INF), 0.);
 }
 
+
+TEST(Mat, template_based_ptr)
+{
+    Mat mat = (Mat_<float>(2, 2) << 0.11f, 0.22f, 0.33f, 0.44f);
+    int idx[2] = {1, 0};
+    ASSERT_DOUBLE_EQ(0.33, *(mat.ptr<float>(idx)));
+    idx[0] = 1;
+    idx[1] = 1;
+    ASSERT_DOUBLE_EQ(0.44, *(mat.ptr<float>(idx)));
```

Comment:
Perhaps we could use integers here to prevent floating-point rounding: `0.xyf` to `xy.0f` (`44.0f`)

Reply:
Oh yes, thanks!